core/src/test/java/com/sap/ai/sdk/core/AiCoreServiceKeyAccessorTest.java

```diff
@@ -0,0 +1,58 @@
+package com.sap.ai.sdk.core;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import com.sap.cloud.environment.servicebinding.api.exception.ServiceBindingAccessException;
+import io.github.cdimascio.dotenv.Dotenv;
+import java.util.Map;
+import org.junit.jupiter.api.Test;
+
+class AiCoreServiceKeyAccessorTest {
+
+  @Test
+  void testValidDotenv() {
+    var dotenv = spy(Dotenv.configure().filename("valid.testenv"));
+
+    var accessor = new AiCoreServiceKeyAccessor(dotenv);
+    assertThat(accessor.getServiceBindings()).isNotEmpty().hasSize(1);
+    assertThat(accessor.getServiceBindings().get(0).getCredentials())
+        .containsEntry("clientid", "cid")
+        .containsEntry("clientsecret", "csecret")
+        .containsEntry("url", "https://authenticationsap")
+        .containsEntry("serviceurls", Map.of("AI_API_URL", "https://api.ai.sap"));
+
+    verify(dotenv, times(1)).load();
```

Comment:
(Minor)

Thanks for adding the test. While this makes sure we're not running I/O heavy task multiple times, the logic still runs `ServiceBinding createServiceBinding(@Nonnull final String serviceKey)` repeatedly. for every AI Core interaction.

Is this accepted / expected?

Reply:
Yeah, also not super happy with that, but considering this is recommended only for non-prod, at least acceptable. I think we can later on add a cache, but then probably rather on the destination object, as caching that skips further object instatiation